src/oled_display_node.cpp

```diff
@@ -578,14 +604,24 @@ int main(int argc, char **argv)
   // Set to subscribe to the display topic and we then get callbacks for each message
   ros::Subscriber sub = nh.subscribe(ROS_TOPIC_DISPLAY_NODE, 1000, displayApiCallback);
 
+  // Set to subscribe to the battery_state topic and we then get callbacks for each message
+  ros::Subscriber sub2 = nh.subscribe("battery_state", 1000, batteryStateApiCallback);
+
+
   // We will refresh the lines from time to time in case IP addr has changed
   ros::Rate loop_rate(0.1);
 
   // mainloop:
   while ((dispInitError == 0) && ros::ok())
   {
-    dispOled_writeText(&oledDisplayCtx, 0, 0, 1, hostname.c_str());
-    dispOled_writeText(&oledDisplayCtx, 2, 0, 1, firstIpAddress.c_str());
+    dispOled_writeText(&oledDisplayCtx, DISP_LINE_HOSTNAME, 0, 1, hostname.c_str());
+    dispOled_writeText(&oledDisplayCtx, DISP_LINE_IP_ADDR, 0, 1, firstIpAddress.c_str());
+
+    std::string battText;
+    std::stringstream stream;
+    stream << std::fixed << std::setprecision(2) << g_batteryVoltage;
+    battText = "BattV: " + stream.str();
```

Comment:
Might as well declare battText down here and initialize it at the same time.

`std::string battText = "BattV:" + steam.str()`